packages/teams-js/test/public/app.spec.ts

```diff
@@ -11,585 +11,1213 @@ import {
+              userClickTime: 2222,
+              userFileOpenPreference: FileOpenPreference.Inline,
+              host: {
+                name: HostName.orange,
+                clientType: HostClientType.web,
+                sessionId: 'someSessionId',
+                ringId: 'someRingId',
+              },
+              appLaunchId: 'appLaunchId',
+            },
+            page: {
+              id: 'someEntityId',
+              frameContext: context,
+              subPageId: 'someSubEntityId',
+              isFullScreen: true,
+              isMultiWindow: true,
+              sourceOrigin: 'www.origin.com',
+            },
+            user: {
+              id: 'someUserObjectId',
+              displayName: 'someTestUser',
+              isCallingAllowed: true,
+              isPSTNCallingAllowed: undefined,
+              licenseType: 'someUserLicenseType',
+              loginHint: 'someLoginHint',
+              userPrincipalName: 'someUserPrincipalName',
+              tenant: { id: 'someTid', teamsSku: 'someTenantSKU' },
+            },
+            channel: {
+              id: 'someChannelId',
+              displayName: 'someChannelName',
+              relativeUrl: 'someChannelRelativeUrl',
+              membershipType: ChannelType.Shared,
+              defaultOneNoteSectionId: 'someDefaultOneNoteSectionId',
+              ownerGroupId: 'someHostGroupId',
+              ownerTenantId: 'someHostTenantId',
+            },
+            chat: { id: 'someChatId' },
+            meeting: { id: 'dummyMeetingId' },
+            sharepoint: {},
+            team: {
+              internalId: 'someTeamId',
+              displayName: 'someTeamName',
+              type: 4,
+              groupId: undefined,
+              templateId: 'someTeamTemplateId',
+              isArchived: false,
+              userRole: 0,
+            },
+            sharePointSite: {
+              teamSiteUrl: 'someSiteUrl',
+              teamSiteDomain: 'someTeamSiteDomain',
+              teamSitePath: 'someTeamSitePath',
+              teamSiteId: 'someSiteId',
+              mySitePath: 'mySitePath',
+              mySiteDomain: 'myDomain',
+            },
           };
-          utils.respondToMessage(message, data.success, data.error);
-          await expect(promise).rejects.toThrowError(mockErrorMessage);
+
+          framelessPostMock.respondToMessage({
+            data: {
+              id: getContextMessage.id,
+              args: [contextBridge],
+            },
+          } as DOMMessageEvent);
+          const actualContext = await contextPromise;
+
+          expect(actualContext).toEqual(expectedContext);
+          expect(actualContext.page.frameContext).toBe(context);
+          expect(actualContext.meeting.id).toBe('dummyMeetingId');
+        });
+      });
+    });
+
+    describe('Testing app.notifyAppLoaded function', () => {
+      it('app.notifyAppLoaded should not allow calls before initialization', () => {
```

Comment:
Should we both with having this test twice? I don't think there is anything different in the setup between framed and frameless, unless I'm missing something?

Reply:
Same with `app.notifySuccess` and `app.notifyFailure`

Reply:
And `app.registerOnThemeChangeHandler`

Reply:
Ah it turns out I'm wrong. There is a difference in setup between framed and frameless and that is the important part of these tests.